### PR TITLE
ndp: provide define for common hop-limit

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
As promised in https://github.com/RIOT-OS/RIOT/pull/10985#discussion_r256117858 here are some documented defines for the hop-limit value for NDP messages.

### Testing procedure
`gnrc_networking` should still compile.

### Issues/PRs references
Follow-up to https://github.com/RIOT-OS/RIOT/pull/10985.

### CODE REVIEW

File: sys/include/net/ndp.h [hunk 1]
```diff
@@ -193,6 +193,18 @@ extern "C" {
 #define NDP_MAX_RANDOM_FACTOR       (1500U)     /**< MAX_RANDOM_FACTOR (x 1000) */
 /** @} */
 
+/**
+ * @brief   Hop-limit required for most NDP messages to ensure link-local
+ *          communitcation
```
Comment (aabadie) [inline sys/include/net/ndp.h]: There is typo here

Comment (miri64) [reply sys/include/net/ndp.h]: Fixed and amended directly.

File: sys/include/net/ndp.h [hunk 2]
```diff
@@ -193,6 +193,18 @@ extern "C" {
 #define NDP_MAX_RANDOM_FACTOR       (1500U)     /**< MAX_RANDOM_FACTOR (x 1000) */
 /** @} */
 
+/**
+ * @brief   Hop-limit required for most NDP messages to ensure link-local
```
Comment (aabadie) [inline sys/include/net/ndp.h]: Why `most` ? It seems that all message types uses this value, no ?

Comment (miri64) [reply sys/include/net/ndp.h]: I'm not so sure about that. Sure, all messages in RFC 4861 use it, but e.g. [DARs and DACs don't use it](https://tools.ietf.org/html/rfc6775#section-4.4).

**Comment by miri64** [comment]
> in debug messages

You want me to do a `printf("%u", NDP_HOP_LIMIT)` instead of `printf("255")`, right? I'm against that.

> in `tests/gnrc_ipv6_nib/main.c` and in `tests/gnrc_ipv6_nib_6ln/main.c`. I think they should be adapted as well.

Done

**Comment by aabadie** [comment]
> You want me to do a `printf("%u", NDP_HOP_LIMIT)` instead of `printf("255")`, right?

Exactly

> I'm against that.

Why? It seems cleaner to me that the debug message takes the expected value from the define.

**Comment by miri64** [comment]
> > You want me to do a `printf("%u", NDP_HOP_LIMIT)` instead of `printf("255")`, right?
> 
> Exactly
> 
> > I'm against that.
> 
> Why? It seems cleaner to me that the debug message takes the expected value from the define.

Because it is adding dynamic code for static configuration (that moreover is not supposed to be changed).

**Comment by kaspar030** [comment]
> Because it is adding dynamic code for static configuration (that moreover is not supposed to be changed).

I know what you mean, but we are talking about DEBUG statements, right? In that case, just imagine someone changing NDP_HOP_LIMIT just to still see "255" in the debug output. The WTF factor will be much higher than the couple of extra cycles needed to convert the number to a string. So I'd say not using the define is premature optimization.

**Comment by kaspar030** [comment]
There's also a macro option:

```
#define STR(x) #x
...
printf(STR(NDP_HOP_LIMIT));
```

**Comment by miri64** [comment]
> There's also a macro option:
> 
> ```
> #define STR(x) #x
> ...
> printf(STR(NDP_HOP_LIMIT));
> ```

Doesn't this just output `NDP_HOP_LIMIT`?

But fine... if we can live with it.

**Comment by kaspar030** [comment]
> Doesn't this just output `NDP_HOP_LIMIT`?

yes, sorry...

**Comment by kaspar030** [comment]
pls squash!

**Comment by miri64** [comment]
Squashed

**Comment by miri64** [comment]
Rebased to current master to include the change in 9216607163f48b805e39874d086e983b0226f0e2 (which `cppcheck` was complaining about). Force-push link reports there are no differences. May I merge once Murdock is done @kaspar030?

**Pull request merged**